deepchecks/nlp/utils/text_properties_models.py

```diff
@@ -0,0 +1,221 @@
+        )
+    )
+
+    if quantize_model is False:
+        model_path = models_storage / 'onnx' / model_name
+
+        if model_path.exists():
+            return onnx.ORTModelForSequenceClassification.from_pretrained(model_path).to(device or -1)
+
+        model = onnx.ORTModelForSequenceClassification.from_pretrained(
+            model_name,
+            export=True,
+            cache_dir=models_storage,
+        ).to(device or -1)
+        # NOTE:
+        # 'optimum', after exporting/converting a model to the ONNX format,
+        # does not store it onto disk we need to save it now to not reconvert
+        # it each time
+        model.save_pretrained(model_path)
+        return model
+
+    model_path = models_storage / 'onnx' / 'quantized' / model_name
+
+    if model_path.exists():
+        return onnx.ORTModelForSequenceClassification.from_pretrained(model_path).to(device or -1)
+
+    not_quantized_model = _get_transformer_model(
+        property_name,
+        model_name,
+        device,
+        quantize_model=False,
+        models_storage=models_storage
+    )
+
+    quantizer = onnx.ORTQuantizer.from_pretrained(not_quantized_model).to(device or -1)
+
+    quantizer.quantize(
+        save_dir=model_path,
+        # TODO: make it possible to provide a config as a parameter
+        quantization_config=onnx.configuration.AutoQuantizationConfig.avx512_vnni(
+            is_static=False,
+            per_channel=False
+        )
+    )
+    return onnx.ORTModelForSequenceClassification.from_pretrained(model_path).to(device or -1)
+
+
+def import_optional_property_dependency(
+        module: str,
+        property_name: str,
+        package_name: Optional[str] = None,
+        error_template: Optional[str] = None
+):
+    try:
+        lib = importlib.import_module(module)
+    except ImportError as error:
+        package_name = package_name or module.split('.', maxsplit=1)[0]
+        error_template = error_template or (
+            'property {property_name} requires the {package_name} python package. '
+            'To get it, run:\n'
+            '>> pip install {package_name}\n\n'
+            'You may install dependencies for all text properties by running:\n'
+            '>> pip install deepchecks[nlp-properties]\n'
+        )
+        raise ImportError(error_template.format(
+            property_name=property_name,
+            package_name=package_name
+        )) from error
+    else:
+        return lib
+
+
+def get_transformer_pipeline(
+        property_name: str,
+        model_name: str,
+        device: Optional[str] = None,
+        models_storage: Union[pathlib.Path, str, None] = None,
+        use_cache=False
+):
+    if use_cache:
+        return _get_transformer_pipeline(property_name, model_name, device, models_storage)
+    return _get_transformer_pipeline.__wrapped__(property_name, model_name, device, models_storage)
+
+
+@lru_cache(maxsize=5)
```

Comment:
why 5?

Reply:
I case that two more will come in the future...